string colorName
int32 offsetX
int32 offsetY
int32 width
int32 height
int32 x
int32 y
int32 radius
